File: ReverseEvenElements/ReverseEvenElements.java
import java.util.*;
import crio.ds.List.ListNode;



class ReverseEvenElements{
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		ArrayList<Integer> inp=new ArrayList<Integer>();
		for(int i=0;i<n;i++)
			inp.add(sc.nextInt());
		sc.close();
		ListNode head=ListNode.createList(inp);
		Solution so=new Solution();
		head=so.reverseEvenElements(head);
		List<Integer> ans=ListNode.extractList(head);
		for(int i=0;i<ans.size();i++)
		{
			System.out.print(ans.get(i)+" ");
		}
	}
}

File: ReverseEvenElements/ReverseEvenElements.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"
#include "../crio/cpp/io/FastIO.hpp"


int main() {
	FastIO();
	int n;
	cin >> n;
	vector<int> list(n);
	for ( int i = 0; i < n; i++) cin >> list[i];
	ListNode *head = ListNode().createList(list);
	head = reverseEvenElements(head, NULL);
	vector<int> result = ListNode().extractList(head);
	for ( int it : result) {
		cout << it << " ";
	}
}
